### PR TITLE
[Keyboard] adds new keyboard version: dztech/dz60rgb/v2_1

### PR DESCRIPTION
**Pull request by trankimtung**
Adds new keyboard version: dztech/dz60rgb/v2_1

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dztech/dz60rgb/v2_1/rules.mk [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   LUFA MS      lufa-ms
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/dztech/dz60rgb/v2_1/rules.mk:12]:
```suggestion
```

File: keyboards/dztech/dz60rgb/v2_1/rules.mk [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
+AUDIO_ENABLE = no              # Audio output on port C6
+RGB_MATRIX_ENABLE = yes        # Use RGB matrix
+RGB_MATRIX_DRIVER = IS31FL3733
+NO_USB_STARTUP_CHECK = no      # Disable initialization only when usb is plugged in
+EXTRAFLAGS += -flto            # Enable link-time optimization
```
Comment (drashna) [inline keyboards/dztech/dz60rgb/v2_1/rules.mk:34]:
```suggestion
LTO_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/dztech/dz60rgb/v2_1/rules.mk [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+CONSOLE_ENABLE = no            # Console for debug
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
+AUDIO_ENABLE = no              # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/dztech/dz60rgb/v2_1/rules.mk:30]:
```suggestion
AUDIO_ENABLE = no              # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/dztech/dz60rgb/readme.md [hunk 4]
```diff
@@ -8,14 +8,17 @@ A hotswap 60% RGB keyboard.
 * Hardware Supported: DZtech DZ60RGB V1 and V2
 * Hardware Availability: [KBDfans](https://kbdfans.com/)
 
-There are two versions of the DZ60RGB. Please use the appropriate firmware for your board.
+There are three versions of the DZ60RGB. Please use the appropriate firmware for your board.
 
 * V1: STM32F303 (Arm), takes .bin files
 * V2: ATmega32U4 (AVR), takes .hex files
+* V2.1: ATmega32U4 (AVR), takes .bin files [Flash Manual](https://docs.google.com/document/d/111qx6Qec4JqtIhWaZlMND-VuRnFtn9a-gJaHN8fsL7M/edit?usp=sharing)
```
Comment (fauxpark) [inline keyboards/dztech/dz60rgb/readme.md:15]: Does KBDfans refer to this as rev2.1?

Comment (trankimtung) [reply keyboards/dztech/dz60rgb/readme.md:15]: No. They refer to it as "new" v2, perhaps to reserve v3 for yet another iteration.

Comment (fauxpark) [reply keyboards/dztech/dz60rgb/readme.md:15]: Ugh, alright then. :)

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**